AzureUtils.sh

```diff
@@ -0,0 +1,212 @@
+  #
+  # per https://learn.microsoft.com/en-us/azure/virtual-machines/linux/create-upload-centos#centos-70
+  #
+  # Currently, the CentOS 7 guidance is the only guidance available. To date,
+  # it's proven to work. This configuration-section will remain the same until
+  # EL8-specific guidance has been released by the Azure team
+
+
+  # If WALinuxAgent RPM is missing, attempt to install from yum-repos
+  if [[ $( rpm -q --quiet WALinuxAgent )$? -ne 0 ]]
+  then
+    STATUS_MSG="The WALinuxAgent RPM is not installed."
+    STATUS_MSG="${STATUS_MSG} Attempting to install."
+
+    err_exit "${STATUS_MSG}" NONE
+    dnf --installroot="${CHROOTMNT}" \
+      install --assumeyes --quiet WALinuxAgent || \
+      err_exit "Failed installing WALinuxAgent" 1
+    err_exit "Success!" NONE
+    
+  fi
+
+  err_exit "Configuring waagent..." NONE
+
+  #  1. Install network-scripts
+  err_exit "Installing RPM dependencies... " NONE
+  dnf --installroot="${CHROOTMNT}" \
+    install --assumeyes --quiet network-scripts || \
+    err_exit "Failed installing RPM-dependencies" 1
+  err_exit "Success!" NONE
+
+  #  2. Enable network.service systemd unit
+  err_exit "Enabling legacy network.service systemd unit " NONE
+  err_exit "in %s... " "${CHROOTMNT}" NONE
+  chroot "${CHROOTMNT}" systemctl enable network.service || \
+    err_exit "Failed enabling network.service" 1
+  err_exit "Success!" NONE
+
+  #  3. Waagent wants an sysconfig/network config file
+  if [[ ! -f ${CHROOTMNT}/etc/sysconfig/network ]]
+  then
+    err_exit "Creating config file for network.service... " NONE
+    install -bDm 0600 -o root -g root /dev/null \
+      "${CHROOTMNT}/etc/sysconfig/network"
+    (
+      echo "NETWORKING=yes"
+      echo "HOSTNAME=localhost.localdomain"
+    ) > "${CHROOTMNT}/etc/sysconfig/network" || \
+      err_exit "Failed creating config file for network.service" 1
+    err_exit "Success!" NONE
+  fi
+
+  #  4. Waagent wants an ifcfg-eth0 config file
+  if [[ ! -f ${CHROOTMNT}/etc/sysconfig/network-scripts/ifcfg-eth0 ]]
+  then
+    err_exit "Creating ifcfg-eth0 file for network.service... " NONE
+    install -bDm 0644 -o root -g root /dev/null \
+      "${CHROOTMNT}/etc/sysconfig/network-scripts/ifcfg-eth0"
+    (
+      echo "DEVICE=eth0"
+      echo "ONBOOT=yes"
+      echo "BOOTPROTO=dhcp"
+      echo "TYPE=Ethernet"
+      echo "USERCTL=no"
+      echo "PEERDNS=yes"
+      echo "IPV6INIT=no"
+      echo "NM_CONTROLLED=no"
+    ) > "${CHROOTMNT}/etc/sysconfig/network-scripts/ifcfg-eth0" || \
+      err_exit "Failed creating ifcfg-eth0 file for network.service" 1
+    err_exit "Success!" NONE
+  fi
+
+  #  5. No static network-naming rules...
+  err_exit "Disabling static udev network-naming rules... " NONE
+  chroot "${AMIGENCHROOT}" ln -s /dev/null \
+    /etc/udev/rules.d/75-persistent-net-generator.rules ||
+    err_exit "Failed disabling static udev network-naming rules" 1
+  err_exit "Success!" NONE
+
+  #  6. Configure waagent for cloud-init
```

Comment:
Can we add a comment here pointing to the README on GitHub? This has the most current info I can find on what options are available and what they do:

```suggestion
  #  6. Configure waagent for cloud-init
  #  For details on waagent config options, see: https://github.com/Azure/WALinuxAgent#configuration-file-options
```